src/video/drawing_context.cpp

```diff
@@ -373,11 +375,13 @@ DrawingContext::do_drawing()
     handle_drawing_requests(lightmap_requests);
     lightmap.end_draw();
 
-    DrawingRequest* request = new(obst) DrawingRequest();
-    request->target = NORMAL;
-    request->type = DRAW_LIGHTMAP;
-    request->layer = LAYER_HUD - 1;
-    drawing_requests.push_back(request);
+    if (!full_brightness) {
+      DrawingRequest* request = new(obst) DrawingRequest();
```

Comment:
While we're at it:
auto request = ...

C'mon, show some respect for C++ 11. :P